File: OOP/OOP_variable/oop_var.py
class C(object):
    def __init__(self, v):
        self.value = v # 인스턴스 변수에 10,v
    def show(self): # 인스턴스를 가르키는 변수
        print(self.value)

c1 = C(10) # v = 10
print(c1.value) # 인스턴스 c1에 소속되어 있는 value 값 출력, 인스턴스 변수의 값을 읽어 온 것.
c1.value = 20 # value 값을 20으로 변경
print(c1.value)
c1.show()

File: OOP/OOP_variable/oop_var.rb
class C # C라는 클래스 생성
    def initialize(v) # __init___ 생성자
        @value = v # @ = self
        #함수 안에서만 인스턴스 변수에 접근 할 수 있다.
    end
    def show()
        p @value # value 값 출력
    end
end

c1 = C.new(10) # v = 10
#p c1.value , 메소드 밖에서는 인스턴스의 값에 영향을 줄 수 없다.
#c1.value = 20
c1.show()
# ruby에서는 인스턴스에 소속 되어있는 인스턴스에 접근하는 것이 안된다.
#p = p()